lib/stack.rb

```diff
@@ -1,19 +1,33 @@
+require_relative './linked_list.rb'
 class Stack
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = LinkedList.new
+    @last_added = nil
+    # raise NotImplementedError, "Not yet implemented"
   end
 
   def push(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @store.nil?
+      @store.add_first(element)
+    else
+      @store.add_last(element)
+    end
+    # raise NotImplementedError, "Not yet implemented"
   end
 
   def pop
-    raise NotImplementedError, "Not yet implemented"
+    @store.remove_last()
```

Comment:
You should probably check to see if the Stack is empty here.

```suggestion
    @store.remove_last() unless @store.empty?
```